tests/testthat/test-merge_sce_list.R

```diff
@@ -580,3 +606,81 @@ test_that("get_altexp_attributes throws an error as expected when features do no
   altExp(sce_list_with_altexp[[1]]) <- altExp(sce_list_with_altexp[[1]])[1:3, ]
   expect_error(get_altexp_attributes(sce_list_with_altexp))
 })
+
+
+test_that("check_metadata throws an error when a field is missing", {
+  metadata(sce_list[[1]])$library_id <- NULL
+  expect_error(check_metadata(sce_list))
+})
+
+test_that("extract_metadata_for_altexp returns the correct values", {
+  expected_list <- list(
+    "library_id" = "library-sce1",
+    "sample_id"  = "sample-sce1"
+  )
+  observed_list <- extract_metadata_for_altexp(sce_list[[1]])
+  expect_equal(
+    names(observed_list), names(expected_list)
+  )
+  expect_equal(
+    unlist(observed_list), unlist(expected_list)
+  )
+})
+
+test_that("prepare_merged_metadata works as expected, with sample_metadata present", {
+  observed_metadata <- sce_list |>
+    purrr::map(metadata) |>
+    prepare_merged_metadata()
+
+  expect_setequal(
+    names(observed_metadata),
+    c("library_id", "sample_id", "library_metadata", "sample_metadata")
+  )
+
+  expect_setequal(
+    observed_metadata$library_id,
+    glue::glue("library-{names(sce_list)}")
+  )
+  expect_setequal(
+    observed_metadata$sample_id,
+    glue::glue("sample-{names(sce_list)}")
+  )
+  expect_setequal(
+    names(observed_metadata$library_metadata),
+    names(sce_list)
+  )
+  expect_setequal(
+    observed_metadata$library_metadata |>
+      purrr::map(names) |>
+      purrr::reduce(intersect),
+    c("library_id", "sample_id", "total_reads")
+  )
+})
+
+
+test_that("prepare_merged_metadata works as expected from altExps metadata (aka, sample_metadata NOT present)", {
+  observed_metadata <- sce_list_with_altexp |>
+    purrr::map(altExp) |>
+    purrr::map(metadata) |>
+    prepare_merged_metadata()
+
+  expect_setequal(
+    names(observed_metadata),
+    c("library_id", "sample_id", "library_metadata")
+  )
+
+  expect_setequal(
+    observed_metadata$library_id,
+    glue::glue("library-{names(sce_list)}")
+  )
+  expect_setequal(
+    observed_metadata$sample_id,
+    glue::glue("sample-{names(sce_list)}")
+  )
+  expect_setequal(
+    observed_metadata$library_metadata |>
+      purrr::map(names) |>
+      purrr::reduce(intersect),
+    c("library_id", "sample_id", "mapped_reads", "ambient_profile")
+  )
+})
```

Comment:
I think these tests are probably redundant. You already do much of the required testing for altExp metadata in the tests for the full merge with altExps, so this is really just repeating the `prepare_merged_metadata()` tests with a slightly different set of data. But it is good that you are testing that the behavior is correct when `sample_metadata` doesn't exist, so I guess maybe leave it!

Reply:
I had almost the identical thought process about redundancy and then deciding maybe it was ok!